From 5af8dd2d8dee59456602c5e5d02651f8655e8ec1 Mon Sep 17 00:00:00 2001
From: Jan Kolarik <jkolarik@redhat.com>
Date: Wed, 28 Jun 2023 11:30:46 +0000
Subject: [PATCH] logger: Use casting to seconds when logging time

Using fmt 10, `time_point<system_clock>` contains seconds represented as a floating-point number. In order to drop the decimal places, we need to explicitly do the conversion. This is also conformant with the currently used fmt-9.1 docs: https://fmt.dev/9.1.0/syntax.html#chrono-format-specifications.
---
 libdnf/logger/logger.cpp | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/libdnf/logger/logger.cpp b/libdnf/logger/logger.cpp
index 9052f4b68..d9bf2fab9 100644
--- a/libdnf/logger/logger.cpp
+++ b/libdnf/logger/logger.cpp
@@ -24,18 +24,18 @@ along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
 
 namespace libdnf {
 
+using namespace std::chrono;
+
 void Logger::log_line(Level level, const std::string & message) noexcept {
-    write(std::chrono::system_clock::now(), getpid(), level, message);
+    write(system_clock::now(), getpid(), level, message);
 }
 
 
 void StringLogger::write(
-    const std::chrono::time_point<std::chrono::system_clock> & time,
-    pid_t pid,
-    Level level,
-    const std::string & message) noexcept {
+    const time_point<system_clock> & time, pid_t pid, Level level, const std::string & message) noexcept {
     try {
-        write(fmt::format("{:%FT%T%z} [{}] {} {}\n", time, pid, level_to_cstr(level), message).c_str());
+        write(fmt::format("{:%FT%T%z} [{}] {} {}\n", time_point_cast<seconds>(time), pid, level_to_cstr(level), message)
+                  .c_str());
     } catch (const std::exception & e) {
         write("Failed to format: ");
         write(message.c_str());
